### ISSUE TITLE
Update debian/ubuntu/etc packages

### ISSUE THREAD

**Issue by revmischa** [projectM-visualizer/projectm#200]
https://packages.debian.org/search?keywords=projectm

These are hopelessly out of date and someone should take over maintenance of the debian packages.

**Comment by Miss-Lonelyhearts** [comment]
Also worth mentioning that the debian package crashes on preset change due to some odd error regarding mislabeled fonts; whoever built the package may have misconfigured it. Do you have an idea of who I should contact to see if the former maintainer could update the package (or where we might find a new person to take it over)? I'm too unskilled to fix it myself it seems, but the project is phenomenal and I'd love to see it running on Debian.

**Comment by coast77777777777** [comment]
Well thanks for taking the time to bring this to our attention. I’m not sure how Debian packages are maintained, but i’d imagine that we can find someone to grant us permissions here. As for the maintainer, I’d be more than happy to help anyone interested taking on that role. It is a great opportunity for a new contributor with any skill level. I would be happy to help walk through all the steps required. 
Thanks,
Coast

**Comment by BrooksPatola** [comment]
anyone take on this role yet? If not may I?

**Comment by revmischa** [comment]
Please do

**Comment by BrooksPatola** [comment]
Anyone willing to help assist with the steps required? 
@coast77777777777

**Comment by rajneesh44** [comment]
Is this issue still open? I am new to open Source and looking forward to contributing.
Thanks in advance if anyone can guide me.

**Comment by revmischa** [comment]
Yes if you're reading this - please feel free to take over maintainership of distribution packaging for whatever OS you're using. Write the current maintainers.
It looks like someone updated to 3.1.3 in Debian, May 2020, which is great: https://ftp-master.debian.org/new/projectm_3.1.3-1.html

**Comment by cr7-ankit** [comment]
Is this issue still open? I am a beginner and  new to open Source and looking forward to contributing. It would be a great help if any of you senior can guide me and make me successful in my very first contribution I am willing to do.
Thanks

**Comment by kblaschke** [comment]
As we're now getting really close to the upcoming 4.0 major release, having someone work on creating Ubuntu/Debian packages for libprojectM and the new SDL frontend would be greatly appreciated.

The project should be structured properly to install the library and development files separately to provide an extra -dev package for those who need headers and the CMake files.

Additionally, the frontend is now a completely separate application and should ideally be treated as such, e.g. install as a separate package. This way, applications only requiring libprojectM.so.4 will only need to install the library. Preset packs may also be packaged separately, but this is up to the package maintainers to decide.

**Comment by spausum** [comment]
Hi, if a maintainer for packages are still needed I would love to help. I have access to linux, windows and macos operating systems. I am new to open-source so would just like a bit of guidance.

**Comment by revmischa** [comment]
Sure, feel free to take on something. Check out the package maintainer guide for whatever you're interested in.

**Comment by kblaschke** [comment]
I second Mischa's opinion! If you're new to maintaining packages, ideally pick one OS/Distro to start with, and familiarize yourself with:
- Building software (build systems, best practices etc.)
- Packaging software for this specific OS, e.g. DEB, RPM, ebuilds and other formats
- The requirements, best practices and caveats of the distro to create, maintain, sign and publish packages
- How to deal with requests, especially security-related reports in a timely and confident manner

Most Linux distros have guides on how to become a maintainer. Here's a short list of the respective pages from some popular distros:
- [Debian](https://wiki.debian.org/DebianMaintainer)
- [Ubuntu](https://wiki.ubuntu.com/NewDevelopersAndMaintainers)
- [Fedora](https://docs.fedoraproject.org/en-US/package-maintainers/Joining_the_Package_Maintainers/)
- [Gentoo](https://www.gentoo.org/get-involved/become-developer/)
- [Arch](https://wiki.archlinux.org/title/Package_Maintainers)
- [openSUSE](https://en.opensuse.org/openSUSE:How_to_contribute_to_Factory)

**One additional note I feel is important enough to state it over and over again:**
Maintaining one or more packages for a single Distro is a _responsible job_, and should only be done if you're _really_ committed to keep yourself involved, e.g. not like "build a package, release it and then move on with other things". no one will easily take over this responsibility and the associated work from you.

**Comment by kblaschke** [comment]
On other operating systems, the landscape is a bit more dull in comparison.

Windows has no standard package repository for libraries. There's vcpkg, for which one of our team members is currently in the process of providing a package (called port here).

For macOS, there is basically only [Homebrew](https://brew.sh/), for which a [projectM formula](https://formulae.brew.sh/formula/projectm#default) already exists, but it is outdated. [The formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/projectm.rb) needs an update to use CMake as the build system, removing the autotools dependencies, checking out sbumodules in Git and update the included build test to use the new C API.

**Comment by spausum** [comment]
> I second Mischa's opinion! If you're new to maintaining packages, ideally pick one OS/Distro to start with, and familiarize yourself with:
> 
>     * Building software (build systems, best practices etc.)
> 
>     * Packaging software for this specific OS, e.g. DEB, RPM, ebuilds and other formats
> 
>     * The requirements, best practices and caveats of the distro to create, maintain, sign and publish packages
> 
>     * How to deal with requests, especially security-related reports in a timely and confident manner
> 
> 
> Most Linux distros have guides on how to become a maintainer. Here's a short list of the respective pages from some popular distros:
> 
>     * [Debian](https://wiki.debian.org/DebianMaintainer)
> 
>     * [Ubuntu](https://wiki.ubuntu.com/NewDevelopersAndMaintainers)
> 
>     * [Fedora](https://docs.fedoraproject.org/en-US/package-maintainers/Joining_the_Package_Maintainers/)
> 
>     * [Gentoo](https://www.gentoo.org/get-involved/become-developer/)
> 
>     * [Arch](https://wiki.archlinux.org/title/Package_Maintainers)
> 
>     * [openSUSE](https://en.opensuse.org/openSUSE:How_to_contribute_to_Factory)
> 
> 
> **One additional note I feel is important enough to state it over and over again:** Maintaining one or more packages for a single Distro is a _responsible job_, and should only be done if you're _really_ committed to keep yourself involved, e.g. not like "build a package, release it and then move on with other things". no one will easily take over this responsibility and the associated work from you.

Hi kblaschke, after some research and reflection, I do realize this is a significant responsibility. As a full time student and someone new to open source, I do not think I am right for the task currently.

**Comment by kblaschke** [comment]
Some of the above distributions already have projectM packages, so if you want to help, you could reach out to the previous maintainers and ask them to update the packages accordingly. There were lots of changes to the project structure though, and the new SDL app is not finished yet, so it's not that we're in a hurry.

**Comment by kblaschke** [comment]
I close this issue as there's a new one which more generally tracks projectM via repology: #774
